src/content/en/2022/pwa.md

```diff
@@ -1,22 +1,517 @@
+sheets_gid="1871711300",
+  sql_file="workbox_usage.sql"
+  )
+}}
+
+Compared with last year we see a big spike in Workbox usage. Last year its usage on mobile was 32.9% compared to 53.64% this year, and almost 60% of PWAs use Workbox in some capacity.
+
+Since the growth that we saw in the number of pages controlled by service workers was not in the top 1000 websites but in more granular categories, and this growth on Workbox usage we can infer that adoption of Workbox is happening inside this companies and websites that might have waited for the technology to be adopted by the top websites, or that might not have the need for a completely custom implementation of service workers and get the most out of Workbox’s tested patterns.
+
+#### Workbox packages
+
+Workbox is structured in a way that developers can choose which parts to add to their projects depending on their site's needs. The usage shown below helps us document which PWA features are developers implementing at the moment.
+
+{{ figure_markup(
+  image="top_workbox_packages.png",
+  caption="Top Workbox packages.",
+  description="Increase in usage of the workbox-core base library.",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vS7KgpDrr_m20ZDaHDBYLFSGNVdj3lhGHbHopEi4071q21U5rZDJfbYrrdyaEIU1D9cwgxxBCI6NBsV/pubchart?oid=205966463&format=interactive",
+sheets_gid="122135540",
+  sql_file="workbox_packages.sql"
+  )
+}}
+
+Here we also see an overall increase in usage, workbox-core. The base library saw an increase of 14% in usage. Workbox-core, together with workbox-routing and workbox-strategies, is used to create a caching strategy that works to serve different artifacts in their webapp to improve performance.  It makes sense they are all at the top as they enable a core benefit of service workers.
+
+There is also a considerable jump in usage on workbox-precaching. Pre-caching can be used to emulate the model that packaged apps use. With workbox-precaching, you can choose assets that will be cached at the time of service worker installation to make those assets load faster in subsequents visits.
+
+What is surprising is the rise in workbox-sw usage,  because starting with [Workbox 5](https://github.com/GoogleChrome/workbox/releases/tag/v5.0.0), the Workbox team has encouraged developers to create custom bundles of the Workbox runtime instead of using `importScripts()` to load <code>[workbox-sw](https://developers.google.com/web/tools/workbox/modules/workbox-sw)</code> (the runtime). The Workbox team will continue supporting <code>workbox-sw</code>, but the new technique is now the recommended approach. In fact, the defaults for the build tools have switched to prefer that method.
+
+It is possible the increase is coming from libraries that use older versions of Workbox like [create-react-app version 3](https://github.com/facebook/create-react-app/blob/v3.4.4/packages/react-scripts/package.json#L82)
+
+
+### Web Push Notifications
+
```

Comment:
@diekus this section should have an intro (no empty space between headings)

Reply:
I've added a small intro to notifications.